site/app/libraries/ExceptionHandler.php

```diff
@@ -20,12 +20,13 @@ class ExceptionHandler {
 
     /**
      * Should we display the actual exception to the user or just a generic message?
-     * This should always be initially true in case we hit an exception in our initial
-     * setup routines
+     * This is initially set to false to prevent leaking any data when loading the config,
+     * but this does mean that by default any exceptions thrown when loading the config
+     * will be swallowed.
      *
      * @var bool
      */
-    private static $display_exceptions = true;
+    private static $display_exceptions = false;
```

Comment:
Essentially this flag tells us whether or not to display an exception to the user or not. This default value only affects the exceptions/errors that are thrown before we've finished loading the config file. By setting it to true, we'll see the actual exception. By setting it to false, we won't.  After the config file is loaded, this value will be overridden by the `$debug` variable in `Config.php`. While there's probably no harm in showing students those exceptions explicitly, I decided to err on the side of caution and hide all exceptions from the public if unsure so to prevent possibly leaking passwords in the future.